stable_baselines3/common/envs/bit_flipping_env.py

```diff
@@ -185,7 +185,7 @@ def step(self, action: Union[np.ndarray, int]) -> GymStepReturn:
         # Episode terminate when we reached the goal or the max number of steps
         info = {"is_success": terminated}
         truncated = self.current_step >= self.max_steps
-        return obs, reward, terminated, truncated, info
+        return cast(GymObs, obs), reward, terminated, truncated, info
```

Comment:
I agree! It is possible to write

```py
obs: Dict[str, np.ndarray] = self._get_obs()  # type: ignore[assignment]
```

which typechecks lines 188-200. But then when returning `obs` it fails, because mypy does not follow the type recursion enough to understand that a `Dict[str, np.ndarray]` is a subtype of `GymObs`. So we have to cast it anyways...

Reply:
And you can't do:
```
obs: GymObs = self._get_obs()  # type: ignore[assignment]
```

Directly? That's super strange. At any rate, deferring to you (also, I weep for my statically typed languages)

Reply:
Well, I can, but then the *following* line errors (with good reason), because:
- some members of the union `GymObs` don't take strings in `__getitem__` (tuple, tensor)
- the values of the  `Dict` GymObs are other `GymObs` which may not be tensors, and the function `compute_reward` wants tensors

Here's the error list in its full glory
```
stable_baselines3/common/envs/bit_flipping_env.py:182: error: No overload variant of "__getitem__" of "tuple" matches argument type "str"  [call-overload]
stable_baselines3/common/envs/bit_flipping_env.py:182: note: Possible overload variants:
stable_baselines3/common/envs/bit_flipping_env.py:182: note:     def __getitem__(self, SupportsIndex, /) -> GymObs
stable_baselines3/common/envs/bit_flipping_env.py:182: note:     def __getitem__(self, slice, /) -> tuple[GymObs, ...]
stable_baselines3/common/envs/bit_flipping_env.py:182: error: Value of type "tuple[GymObs, ...] | dict[str, GymObs] | ndarray[Any, Any] | int" is not indexable  [index]
stable_baselines3/common/envs/bit_flipping_env.py:182: error: Argument 1 to "compute_reward" of "BitFlippingEnv" has incompatible type "Any | tuple[GymObs, ...] | dict[str, GymObs] | ndarray[Any, Any] | int | ndarray[Any, dtype[Any]]"; expected "int | ndarray[Any, Any]"  [arg-type]
stable_baselines3/common/envs/bit_flipping_env.py:182: error: Argument 2 to "compute_reward" of "BitFlippingEnv" has incompatible type "Any | tuple[GymObs, ...] | dict[str, GymObs] | ndarray[Any, Any] | int | ndarray[Any, dtype[Any]]"; expected "int | ndarray[Any, Any]"  [arg-type]
```

Reply:
Ahhhhhh, thank you. I missed that obs was being used elsewhere in between.